frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/storage-efficiency-card/storage-efficiency-card.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import {
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import { PrometheusResponse } from '@console/internal/components/graphs';
+
+import {
+  POOL_STORAGE_EFFICIENCY_QUERIES,
+  StorageDashboardQuery,
+} from '../../../../constants/queries';
+import { EfficiencyItemBody } from './storage-efficiency-card-item';
+import { getGaugeValue } from '../../../../utils';
+
+import './storage-efficiency-card.scss';
+import { humanizeBinaryBytes } from '@console/internal/components/utils';
+
+const StorageEfficiencyCard: React.FC<DashboardItemProps> = ({
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+}) => {
+  React.useEffect(() => {
+    Object.keys(POOL_STORAGE_EFFICIENCY_QUERIES).forEach((key) =>
+      watchPrometheus(POOL_STORAGE_EFFICIENCY_QUERIES[key]),
+    );
+    return () =>
+      Object.keys(POOL_STORAGE_EFFICIENCY_QUERIES).forEach((key) =>
+        stopWatchPrometheusQuery(POOL_STORAGE_EFFICIENCY_QUERIES[key]),
+      );
+  }, [watchPrometheus, stopWatchPrometheusQuery]);
+
+  const poolCapacityRatioResult = prometheusResults.getIn([
+    POOL_STORAGE_EFFICIENCY_QUERIES[StorageDashboardQuery.POOL_CAPACITY_RATIO],
+    'data',
+  ]) as PrometheusResponse;
+
+  const poolCapacityRatioResultError = prometheusResults.getIn([
+    POOL_STORAGE_EFFICIENCY_QUERIES[StorageDashboardQuery.POOL_CAPACITY_RATIO],
+    'loadError',
+  ]);
+
+  const poolSavedResult = prometheusResults.getIn([
+    POOL_STORAGE_EFFICIENCY_QUERIES[StorageDashboardQuery.POOL_SAVED_CAPACITY],
+    'data',
+  ]) as PrometheusResponse;
+
+  const poolSavedResultError = prometheusResults.getIn([
+    POOL_STORAGE_EFFICIENCY_QUERIES[StorageDashboardQuery.POOL_SAVED_CAPACITY],
+    'loadError',
+  ]);
+
+  const ratio = getGaugeValue(poolCapacityRatioResult);
+  const saved = getGaugeValue(poolSavedResult);
+
+  const compressionStats = () => {
+    const capacityRatio = Number(ratio);
+    return capacityRatio === 0 ? '1:1' : `${Math.round(capacityRatio)}:1`;
+  };
+
+  const savingStats = () => {
+    const savingsValue = Number(saved);
+    const savedBytes = humanizeBinaryBytes(savingsValue).string;
+    return savingsValue === 0 ? 'No Savings' : savedBytes;
+  };
```

Comment:
When `getGaugeValue(poolCapacityRatioResult)` returns `undefined`.
Then you will have `capacityRatio`  and `savingsValue` as `0` and it will render a text `1:1` and `No savings`.
You dont want to show that , instead you want to have `Not available`.

You can have :
```
const compressionStats = () => {
   if (capacityRatio) {
       const capacityRatio = Number(ratio);
       return capacityRatio === 0 ? '1:1' : `${Math.round(capacityRatio)}:1`;
    };
     return capacityRatio;
  };

  const savingStats = () => {
      if(savingsValue) {
          const savingsValue = Number(saved);
          const savedBytes = humanizeBinaryBytes(savingsValue).string;
          return savingsValue === 0 ? 'No Savings' : savedBytes;
       }
       return savingsValue;
  };
```

Reply:
@afreen23 if `getGaugeValue(poolCapacityRatioResult)` returns `undefined`,  `capacityRatio` and `ratio` will be `NaN`, becasue `Number(undefined)` gives `NaN` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number.

Therefore, we dont need to put if, because I have checking for value to be 0 or not. wdyt?

Reply:
@a2batic  you are missing my point here. What I pointed above is if `capacityRatio` and `savingsValue` wont be `0` and they will default to the text. Instead I would at least prefer to set for such cases `Not available`.
An improvement over `stats ? handler() : null;` , we can avoid explicitly setting `null` and let the handler itself respond.

I am alright if you want to keep it that way.

Reply:
@afreen23, thanks for clarifying, I understand your point, only reasons to put it in element, So we dont have to check for every item in their respective functions.

Reply:
What about doing this way https://github.com/openshift/console/pull/6104#pullrequestreview-456930993 ?

Reply:
ack